modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java

```diff
@@ -5895,14 +5895,24 @@ private boolean _hasFinderThatIsNotUniqueByCompany(
 			return false;
 		}
 
-		List<String> finderColumnNames = ListUtil.filter(
-			TransformUtil.transform(
-				finderColumnElements,
-				finderColumnElement -> finderColumnElement.attributeValue(
-					"name")),
-			finderColumnName ->
-				finderColumnName.endsWith("Id") ||
-				finderColumnName.endsWith("PK"));
+		List<String> finderColumnNames = TransformUtil.transform(
+			finderColumnElements,
+			finderColumnElement -> {
+				String finderColumnName = finderColumnElement.attributeValue(
+					"name");
+
+				if (finderColumnName == null) {
+					return null;
+				}
+
+				if (finderColumnName.endsWith("Id") ||
```

Comment:
@slnn Please flip this `if`, we usually put `return null` in shortcut.

Reply:
Done